pkg/instance/instance.go

```diff
@@ -282,25 +277,10 @@ func (i *Instance) SetGitRepo(ctx context.Context, gitContext builder.GitContext
 	return i.builderFactory.BuildImageFromGitRepo(ctx, gitContext, imageName)
 }
 
-// SetImageInstant sets the image of the instance without a grace period.
-// Instant means that the pod is replaced without a grace period of 1 second.
-// It is only allowed in the 'Running' state.
-func (i *Instance) SetImageInstant(ctx context.Context, image string) error {
-	if !i.IsState(StateStarted) {
-		return ErrSettingImageNotAllowedForSidecarsStarted.WithParams(i.state.String())
-	}
-
-	if i.isSidecar {
-		return ErrSettingImageNotAllowedForSidecars
-	}
-
-	return i.setImageWithGracePeriod(ctx, image, nil)
-}
```

Comment:
When calling `SetImage` in `StateStarted`, it uses a grace period by default. There is a use case where you want to set this grace period to 0. This is why this function was introduced.

Reply:
If we would introduce a function like `UpgradeImage`, we could let the user pass an optional parameter to allow to set the grace period of upgrading the image.